### DISCUSSION TITLE
How to expose only get endpoints per Router

### DISCUSSION THREAD

**Discussion by junoriosity** [fastapi/fastapi#5619] [Questions] [answered]
I have a several Routers whose endpoints I expose on a single app. These endpoints can be anything from post, get, put, delete.

This works nicely, but I am wondering whether there is an elegant solution to just expose the get endpoints for routers?

Of course, I could wrap every endpoint definition in an if condition, but perhaps there is a better way to do it. 😀

**Comment by tiangolo** [comment, marked as answer]
Probably better to ask in an issue, following the template and creating a minimal self-contained example that shows your use case. That will make it easier to understand your needs and help you.

**Comment by junoriosity** [comment]
@tiangolo Many thanks for your input. I have created a [new issue](https://github.com/tiangolo/fastapi/issues/5620).

**Reply by tiangolo** [reply]
Thanks!
